### ISSUE TITLE
Code shouldn’t unnecessarily redefine LOG_GENERAL

### ISSUE THREAD

**Issue by cuavas** [mamedev/mame#10183]
logmacro.h itself defines `LOG_GENERAL` to 1 (aka `(1U << 0)`) and uses it with the abbeviated `LOG` macro (as opposed to `LOGMASKED`) for convenience.  Due to the way the preprocessor works, it’s safe to use `LOG_GENERAL` when defining `VERBOSE` before including logmacro.h itself.

In the interest of keeping things familiar, unless there’s a very good reason, code should not:
* Define some other log constant to `1`, `(1U << 0)` or equivalent.
* Redefine `LOG_GENERAL`, especially to values other that `1`.
* Redefine `LOG`.

**Comment by cuavas** [comment]
Opening at the request of @MooglyGuy.

**Comment by MooglyGuy** [comment]
Thanks, replying so that hopefully Github will let you assign it to me now.

Question: With "don't redefine `LOG`", something like this would still be acceptable, yes?

`#define LOGREGISTER(...) LOGMASKED(LOG_REGISTERS, __VA_ARGS__)`

**Comment by cuavas** [comment]
Yes, that’s fine, just don’t redefine `LOG` itself to do something weird.
